go/enclave/rpcencryptionmanager/rpc_encryption_manager.go

```diff
@@ -140,9 +139,12 @@ func (rpc *RPCEncryptionManager) DecryptTx(encryptedTx common.EncryptedTx) (*com
 		return nil, fmt.Errorf("could not decrypt transaction with enclave private key. Cause: %w", err)
 	}
 
-	// We need to extract the transaction hex from the JSON list encoding. We remove the leading `"[0x`, and the trailing `]"`.
-	txBinary := txBinaryListJSON[4 : len(txBinaryListJSON)-2]
-	txBytes := gethcommon.Hex2Bytes(string(txBinary))
+	var txList []string
+	err = json.Unmarshal(txBinaryListJSON, &txList)
+	if err != nil || txList == nil || len(txList) == 0 {
```

Comment:
I think we should give a different error for the final condition - `len(txList) == 0`. That implies that you submitted nothing, not that there was some extraction failure.